extensions/amp-app-banner/1.0/component/ios.js

```diff
@@ -0,0 +1,106 @@
+import {docService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+
+import {user, userAssert} from '#utils/log';
+
+import {openWindowDialog} from '../../../../src/open-window-dialog';
+
+const OPEN_LINK_TIMEOUT = 1500;
+
+/**
+ * @return {{
+ *    openOrInstall: function(): void,
+ *    installAppUrl: string,
+ *    openInAppUrl: string,
+ * }|null}
+ */
+export function getIOSAppInfo() {
+  const canShowBuiltinBanner =
+    !viewerService.isEmbedded() && platformService.isSafari();
+  if (canShowBuiltinBanner) {
+    user().info(
+      'BENTO-APP-BANNER',
+      'Browser supports builtin banners. Not rendering amp-app-banner.'
+    );
+    return null;
+  }
+
+  const canNavigateTo =
+    !viewerService.isEmbedded() || viewerService.hasCapability('navigateTo');
+  if (!canNavigateTo) {
+    return null;
+  }
+
+  const metaContent = docService.getMetaByName('apple-itunes-app');
+  if (!metaContent) {
+    return null;
+  }
+  const {installAppUrl, openInAppUrl} = parseIOSMetaContent(metaContent);
+  if (!installAppUrl) {
+    return null;
+  }
+
+  return {
+    installAppUrl,
+    openInAppUrl,
+    openOrInstall: () => {
+      if (viewerService.isEmbedded()) {
+        user().warn(
+          'BENTO-APP-BANNER',
+          'Bento components should never be running in an embedded viewer'
+        );
+        // timerService.delay(() => {
+        //   viewerService.sendMessage('navigateTo', dict({'url': installAppUrl}));
+        // }, OPEN_LINK_TIMEOUT);
+        // viewerService.sendMessage('navigateTo', dict({'url': openInAppUrl}));
+      } else {
+        timerService.delay(() => {
+          openWindowDialog(window, installAppUrl, '_top');
+        }, OPEN_LINK_TIMEOUT);
+        openWindowDialog(window, openInAppUrl, '_top');
+      }
+    },
+  };
+}
+
+/**
+ * @param {string} metaContent
+ * @return {{installAppUrl: string, openInAppUrl: string}}
+ */
+export function parseIOSMetaContent(metaContent) {
+  const parts = metaContent.replace(/\s/, '').split(',');
+  const config = {};
+  parts.forEach((part) => {
+    const keyValuePair = part.split('=');
+    config[keyValuePair[0]] = keyValuePair[1];
+  });
```

Comment:
This would probably be a good use for a `reduce`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

```suggestion
  const config = parts.reduce((accumulator, part) => {
      const [key, value] = part.split('=');
      return {
          ...accumulator,
          [key]: value
      }
  }, {})
```

Reply:
Done (again, in both `1.0` and `0.1`, to keep the symmetry)